### PR TITLE
Remove Theme Data on Theme Deletion

### LINKED ISSUE #1292: [Bug?] Theme data not deleted after theme deleted
Hi,

When I delete a Theme via the "Manage" dropdown in the `Settings > CMS > Front-End Theme` section, the related theme data in the DB table `cms_theme_data` isn't removed - should it be?

### PR DESCRIPTION
**Pull request by DanHarrin**
Fixes #1292, the second oldest issue on this repo 😄

For reference, custom theme data that lives in `cms_theme_data` is deleted with the theme.

### CODE REVIEW

File: modules/cms/classes/Theme.php [hunk 1]
```diff
@@ -524,6 +524,17 @@ public function getCustomData()
...
+    /**
+     * Remove data specific to this theme
+     * @return bool
+     */
+    public function removeCustomData()
+    {
+        if ($this->hasCustomData()) {
+            return $this->getCustomData()->delete();
+        }
+    }
```
Comment (bennothommo) [inline modules/cms/classes/Theme.php]: @DanHarrin please add a `return true` at the bottom of this method, as the docblock indicates a boolean return value.

**Comment by DanHarrin** [comment]
All done @bennothommo :)

**Comment by bennothommo** [comment]
Nice one, @DanHarrin!

**Comment by w20k** [comment]
Tested - works great ;)

**Comment by LukeTowers** [comment]
@DanHarrin is there currently a warning message before a theme deletion is processed that indicates that the theme files and all theme data will be removed?

**Comment by DanHarrin** [comment]
@LukeTowers yes, there is

**Pull request merged**